src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -15,6 +17,125 @@ namespace System.Net.Http
 {
     internal static class ConnectHelper
     {
+        public static ValueTask<Stream> ConnectAsync(string host, int port, bool async, CancellationToken cancellationToken)
+        {
+            return async ? ConnectAsync(host, port, cancellationToken) : new ValueTask<Stream>(Connect(host, port, cancellationToken));
+        }
+
+        private static async ValueTask<Stream> ConnectAsync(string host, int port, CancellationToken cancellationToken)
```

Comment:
This is all concerning to me.  The premise of the SocketsHttpHandler.ConnectCallback's design (and not exposing the default implementation of how a socket is created or the default connect logic) was that it's just a couple of lines of code to emulate the default behavior... now it's over a 100 lines?
cc: @geoffkizer

Reply:
We're going to introduce an easy mode static `Socket.ConnectAsync` API, so it'll go back to a 5 liner.

Reply:
Then shouldn't we add that and then use it here rather than doing it this way? I don't see why we're putting back this connect helper logic.  If the concern is being able to more easily backport, that same static API can just be added as an internal SocketEx static in the backport or something like that.

Reply:
I agree with @stephentoub here